### PR TITLE
Hacking to enable HTTP auth with PHP-CGI + Apache suExec.

### PR DESCRIPTION
**Pull request by filipesaraiva**
The hacking was proposed by Matthias Blaicher (http://ur1.ca/jfz9v) to access
owncloud in Webfaction servers. More informations in Webfaction community
support http://ur1.ca/jg37x.

### CODE REVIEW

**Comment by filipesaraiva** [comment]
This pull request is related to issue #10318.

File: lib/base.php [hunk 1]
```diff
@@ -841,6 +841,13 @@ protected static function handleAuthHeaders() {
 				break;
 			}
 		}
+
+		// Set HTTP auth headers for HTTP Authentication with PHP-CGI
+		if(isset($_GET['Authorization']) && preg_match('/Basic\s+(.*)$/i', $_GET['Authorization'], $matches)) {
```
Comment (LukasReschke) [inline lib/base.php]: I'm honestly against doing some magical hacks that involves parsing all `Authorization` GET parameters globally in base.php

**Comment by LukasReschke** [comment]
That hack should get implemented as own application instead of some magical base.php hack that might very likely lead to problems in other environments.

You should create a new application and add that hack to it's app.php – from my PoV this is nothing that belongs in core.

**Comment by mabl** [comment]
Hi,
I'm the guy who wrote the original blog post. 

While I think the solution is safe from a security point of view, it's rather invasive for all other deployments which is why I never pursuited to push it upstream. 

@LukasReschke Could this actually be done in applications? It would have to modify the way authentication work and also change .htaccess. Without ever having looked into Owncloud apps, I would not have expected them to be that invasive.

Regarding licences: Since most of the solution was already circulating around previously on the net I'd consider it MIT or WTFPL.

Best,
Matthias

**Comment by filipesaraiva** [comment]
Hello, I just created the merge request after the request from @DeepDiver1975. Well, is it possible to do it as an app? If yes, please send me some information about app development to owncloud and I will try to do it. Really, I would like stop to apply this patch to each update of my owncloud instance. =) Thanks.

**Comment by LukasReschke** [comment]
Let me reply to that later… – It's definitely to do that as app.

**Comment by MorrisJobke** [comment]
ping

**Comment by MorrisJobke** [comment]
> Let me reply to that later… – It's definitely possible to do that as app.

@LukasReschke Yes?

**Comment by simonv3** [comment]
:+1: Would love some ideas on how to do this in an app. We could include that here: https://github.com/owncloud/core/issues/10318

**Comment by DeepDiver1975** [comment]
@LukasReschke will come up with an app to handle this ...

**Comment by filipesaraiva** [comment]
Thanks, but please could you ping this bugreport when the app be released?
Cheers!

**Comment by filipesaraiva** [comment]
Any news about the app to fix it?
Let me know if I can help with something.

**Pull request closed without merging**